### ISSUE TITLE
The application may be doing too much work on its main thread.

### ISSUE THREAD

**Issue by syed-rafiul-hussain** [schwabe/ics-openvpn#403]
0-13 08:02:08.453 4458-15498/de.blinkt.openvpn I/OpenVPN: Starting openvpn
10-13 08:02:08.475 4458-4458/de.blinkt.openvpn I/Choreographer: Skipped 128 frames!  The application may be doing too much work on its main thread.
10-13 08:02:08.525 4458-15498/de.blinkt.openvpn I/OpenVPN: Giving up
10-13 08:02:08.573 4458-15498/de.blinkt.openvpn I/OpenVPN: Exiting

**Comment by schwabe** [comment]
Yes I know it is but the cause for that is totally unknown so it will stay that way.

**Comment by syed-rafiul-hussain** [comment]
On the app's log I have found the following message:

PM ERROR: Endtag <\/ca> missing 
Exiting due to fatal error.

The apk downloaded from play store runs smoothly without having any such error.

**Comment by wvrfish** [comment]
I noticed this UI lockup and tracked it down to OpenVpnManagementThread.java.

On line 67 there is a loop that binds `mServerSocketLocal` which seems to throw an exception every iteration after the first (each with a `Thread.sleep`). I think the condition for the `while` should be `!mServerSocketLocal.isBound()` instead of `!mServerSocketLocal.isConnected()`. I might be wrong as I don't fully understand the socket process but I tried it on my machine and the freeze seems to disappear.
